crates/proto/src/xfer/dns_multiplexer.rs

```diff
@@ -215,20 +213,18 @@ where
     }
 
     /// creates random query_id, validates against all active queries
-    fn next_random_query_id(&self) -> Async<u16> {
+    fn next_random_query_id(&self) -> Poll<u16> {
         let mut rand = rand::thread_rng();
 
         for _ in 0..100 {
             let id: u16 = Standard.sample(&mut rand); // the range is [0 ... u16::max]
 
             if !self.active_requests.contains_key(&id) {
-                return Async::Ready(id);
+                return Poll::Ready(id);
             }
         }
 
-        warn!("could not get next random query id, delaying");
-        task::current().notify();
-        Async::NotReady
+        Poll::Pending
```

Comment:
isn't this gonna bubble up the pending, and never going to be called back again? (scheduler is not going to know when this is to become ready again)

Reply:
oh, thanks for catching this. I need to pass through the Context or a Waker to immediately wake on this.

Reply:
ok, this should be covered now.